core/state/statedb_test.go

```diff
@@ -1373,3 +1373,221 @@ func TestStorageDirtiness(t *testing.T) {
 	state.RevertToSnapshot(snap)
 	checkDirty(common.Hash{0x1}, common.Hash{0x1}, true)
 }
+
+func TestCheckTransactionConditional(t *testing.T) {
+	type preAction struct {
+		Account common.Address
+		Slots   map[common.Hash]common.Hash
+	}
+
+	tests := []struct {
+		name       string
+		preActions []preAction
+		cond       types.TransactionConditional
+		valid      bool
+	}{
+		{
+			// Clean Prestate, no defined cond
+			name:       "clean prestate",
+			preActions: []preAction{},
+			cond:       types.TransactionConditional{},
+			valid:      true,
+		},
```

Comment:
It should be possible to omit the variable names for an implied structure like this, which might make the test structure smaller. But maybe not easier to understand.

Reply:
yea my editor yells at me that these variables don't have to be named but it makes the tests cleaner imo